native/src/main/java/io/ballerina/stdlib/io/nativeimpl/ToTable.java

```diff
@@ -149,6 +151,10 @@ private static void populateRecord(int type, Map<String, Object> struct, String
             case TypeTags.FLOAT_TAG:
                 struct.put(fieldName, (value == null || value.isEmpty()) ? null : Double.parseDouble(value));
                 break;
+            case TypeTags.DECIMAL_TAG:
+                struct.put(fieldName, (value == null || value.isEmpty()) ? null :
+                        ValueCreator.createDecimalValue(value.replaceAll(" ", "")));
```

Comment:
why do we need to replace spaces here? if it is a decimal type, there should not be any spaces in value, right?

Reply:
@sahanHe we can trim the value if you want to remove any extra spaces at the beginning or end?

Reply:
Replace was introduced to remove the possible spaces present in comma separated values since it will result in errors. I will look into trim as an alternative.